### PR TITLE
label more of allready known values

### PR DESCRIPTION
**Pull request by zelosos**

### CODE REVIEW

File: src/constants/directions.asm [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
 ; - wRoomTransitionDirection
-DIRECTION_RIGHT EQU $00
-DIRECTION_LEFT  EQU $01
-DIRECTION_UP    EQU $02
-DIRECTION_DOWN  EQU $03
+DIRECTION_RIGHT     EQU $00
+DIRECTION_LEFT      EQU $01
+DIRECTION_UP        EQU $02
+DIRECTION_DOWN      EQU $03
+DIRECTION_INVALIDE  EQU $0F
```
Comment (kemenaran) [inline src/constants/directions.asm:8]: Minor typo:

```suggestion
DIRECTION_INVALID   EQU $0F
```

File: src/code/bank2.asm [hunk 2]
```diff
@@ -1558,11 +1583,21 @@ func_002_4B49::
...
+.up:    db  $08
+.down:  db  $08
+
+; Convert the view direction of Link to an index offset in Y direction.
+; used for sword collision detection
+LinkDirectionToAdjacentTileIndexY::
+.right: db  $0A
+.left:  db  $0A
+.up:    db  $FC
+.down:  db  $14
```
Comment (kemenaran) [inline src/code/bank2.asm:1600]: Nice finding 👍

File: src/code/bank2.asm [hunk 3]
```diff
@@ -2804,53 +2844,66 @@ func_002_52D6::
...
-
-Data_002_5300::
-    db   $02, $02, $22, $22, $22, $02, $02, $42, $22, $02, $02, $22, $02, $02, $22, $22
-
-label_002_5310:
+; convert Link's view direction to ???
+LinkDirectionTohMultiPurpose1::
+    ;   right   left    up      down
+    db  $0D,    $F3,    $00,    $FF
+    db  $08,    $F8,    $0C,    $F5
```
Comment (kemenaran) [inline src/code/bank2.asm:2851]: Nice formatting, it makes the right/left/up/down values quite readable 👌

File: src/code/bank20.asm [hunk 4]
```diff
@@ -1060,20 +1081,28 @@ func_020_4B4A::
...
+.right: db   0
+.left:  db   0
+.up:    db  -3
+.down:  db   4
 
 func_020_4B81::
     ld   hl, wEntitiesPrivateCountdown1Table      ; $4B81: $21 $F0 $C2
     add  hl, de                                   ; $4B84: $19
     ld   [hl], $10                                ; $4B85: $36 $10
+    ; reset wBombArrowCooldown if not allready zero
```
Comment (kemenaran) [inline src/code/bank20.asm:1102]: Minor typo:
 
```suggestion
    ; reset wBombArrowCooldown if not already zero
```

File: src/constants/gbhw.asm [hunk 5]
```diff
@@ -32,14 +32,18 @@ SERIAL   EQU 3
...
+J_UP     EQU 1 << 2     ;  4
+J_DOWN   EQU 1 << 3     ;  8
+J_A      EQU 1 << 4     ; 10
+J_B      EQU 1 << 5     ; 20
+J_SELECT EQU 1 << 6     ; 40
+J_START  EQU 1 << 7     ; 80
+J_DIRECTION_MASK            EQU J_RIGHT | J_LEFT | J_UP | J_DOWN ; $0F: filter A, B, Select and Start out
+J_DIRECTION_HORIZONTAL_MASK EQU J_RIGHT | J_LEFT                 ; $03: only keep right and left
+J_DIRECTION_VERTICAL_MASK   EQU J_UP | J_DOWN                    ; $0C: only keep up and down
+J_AB_MASK                   EQU J_A | J_B                        ; $30: only keep A and B
```
Comment (kemenaran) [inline src/constants/gbhw.asm:46]: To be honest, in the actual code, I find:

```suggestion
  and (J_UP | J_DOWN)
```

to be easier to read than:

```suggestion
  and J_DIRECTION_VERTICAL_MASK
```

But if you feel having more constants is better, that's also fine with me.

File: src/constants/status.asm [hunk 6]
```diff
@@ -0,0 +1,13 @@
+;
+; states that a varaible for internal use can get
```
Comment (kemenaran) [inline src/constants/status.asm:2]: Nitpick: IMO we can name this file `constants/collisions.asm`.  We'll probably have several collision-related constants to put in it later.

**Review by kemenaran — APPROVED** [review]
That's quite a lot of work. I think it's a good idea you have to create more constants for more values; it makes the code easier to read and more organized 👍 

Other than a few minor nitpicks, that's good for me.

File: src/constants/directions.asm (hunk 1 above)
Comment (zelosos) [reply src/constants/directions.asm:8]: could you fix the typos?

File: src/constants/gbhw.asm (hunk 5 above)
Comment (zelosos) [reply src/constants/gbhw.asm:46]: Ok was an idea. Was not sure about that. But I would remove the brackets. They add no value.

File: src/constants/status.asm (hunk 6 above)
Comment (zelosos) [reply src/constants/status.asm:2]: sounds reasonable. Also some constants in gameplay belong in a different file. My naming for that is not the best, so I let this work do someone else.

File: src/constants/gbhw.asm (hunk 5 above)
Comment (zelosos) [reply src/constants/gbhw.asm]: replaced again

File: src/constants/directions.asm (hunk 1 above)
Comment (kemenaran) [reply src/constants/directions.asm]: Sure; done.

File: src/constants/status.asm (hunk 6 above)
Comment (kemenaran) [reply src/constants/status.asm]: Allright. I renamed the file, and pushed it to this PR.

**Comment by kemenaran** [comment]
@zelosos I added two new commits, to rename the `DIRECTION_INVALID` constant, and to rename `constants/status.asm` to `constants/physics.asm`.

If this is fine with you, you can merge this PR yourself 👍

**Pull request merged**